### ISSUE TITLE
`const` without a type fails at runtime but is allowed statically

### ISSUE THREAD

**Issue by aisamanra** [sorbet/sorbet#2790]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20A%20%3C%20T%3A%3AStruct%0A%20%20const%20%3Ax%0Aend%0A)

```ruby
class A < T::Struct
  const :x
end
```

#### Observed output

From the typechecker:
```
No errors! Great job.
```

Upon loading the code:
```
[path]/lib/types/private/methods/signature.rb:166:in `each_args_value_type`: wrong number of arguments (given 1, expected 2..3) (ArgumentError)
```

#### Expected behavior

Sorbet should disallow it statically.

- - -

**Comment by jez** [comment]
This is not specific to `const`, it happens with `prop` too.

**Comment by aisamanra** [comment]
This is fixed by #2792.
